### PR TITLE
Add smoke test for websphere, update smoke test server versions

### PR DESCRIPTION
**Pull request by laurit**
Continuation of https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/4598

### CODE REVIEW

File: instrumentation/liberty/liberty/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/liberty/LibertyWebAppInstrumentation.java [hunk 1]
```diff
@@ -52,9 +51,15 @@ public void transform(TypeTransformer transformer) {
     @Advice.OnMethodEnter(suppress = Throwable.class)
     public static void onEnter(
         @Advice.Argument(value = 0) ServletRequest request,
-        @Advice.Argument(value = 1) ServletResponse response) {
-
-      if (!(request instanceof HttpServletRequest) || !(response instanceof HttpServletResponse)) {
+        @Advice.Argument(value = 1) ServletResponse response,
+        @Advice.Local("otelContext") boolean handled) {
```
Comment (trask) [inline instrumentation/liberty/liberty/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/liberty/LibertyWebAppInstrumentation.java:55]:
```suggestion
        @Advice.Local("otelHandled") boolean handled) {
```

File: instrumentation/liberty/liberty/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/liberty/LibertyWebAppInstrumentation.java [hunk 2]
```diff
@@ -69,7 +74,11 @@ public static void onEnter(
     public static void stopSpan(
         @Advice.Argument(0) ServletRequest servletRequest,
         @Advice.Argument(1) ServletResponse servletResponse,
-        @Advice.Thrown Throwable throwable) {
+        @Advice.Thrown Throwable throwable,
+        @Advice.Local("otelContext") boolean handled) {
```
Comment (trask) [inline instrumentation/liberty/liberty/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/liberty/LibertyWebAppInstrumentation.java:78]:
```suggestion
        @Advice.Local("otelHandled") boolean handled) {
```

File: smoke-tests/src/test/groovy/io/opentelemetry/smoketest/JettyJpmsSmokeTest.groovy [hunk 3]
```diff
@@ -16,7 +16,7 @@ abstract class JettyJpmsSmokeTest extends AppServerTest {
   @Override
   protected String[] getCommand() {
     // --jpms flags enables using java module system
-    return ["java", "-jar", "/server/start.jar", "--jpms"]
+    return ["java", "-jar", "/server/start.jar", "--jpms", "-Dcom.sun.management.jmxremote"]
```
Comment (trask) [inline smoke-tests/src/test/groovy/io/opentelemetry/smoketest/JettyJpmsSmokeTest.groovy:19]: just checking that this is an intentional change (no objection if it is)

File: docs/supported-libraries.md [hunk 4]
```diff
@@ -119,11 +119,12 @@ These are the supported application servers:
...
+| [Weblogic](https://www.oracle.com/java/weblogic/)                                         | 12.x                        | Oracle JDK 8     | Oracle Linux 7, 8              |
+| [Weblogic](https://www.oracle.com/java/weblogic/)                                         | 14.x                        | Oracle JDK 8, 11 | Oracle Linux 7, 8              |
+| [Websphere Liberty Profile](https://www.ibm.com/cloud/websphere-liberty)                  | 20.x, 21.x                  | OpenJDK 8, 11    | Ubuntu 18, Windows Server 2019 |
+| [Websphere Traditional](https://www.ibm.com/cloud/websphere-application-server)           | 8.5.x, 9.0.x                | IBM JDK 8        | Red Hat Enterprise Linux 8.4   |
+| [WildFly](https://www.wildfly.org/)                                                       | 13.x                        | OpenJDK 8        | Ubuntu 18, Windows Server 2019 |
+| [WildFly](https://www.wildfly.org/)                                                       | 17.x, 21.x, 25.x            | OpenJDK 8, 11    | Ubuntu 18, Windows Server 2019 |
```
Comment (trask) [inline docs/supported-libraries.md:127]: 👍

**Review by trask — APPROVED** [review]

File: smoke-tests/src/test/groovy/io/opentelemetry/smoketest/JettyJpmsSmokeTest.groovy (hunk 3 above)
Comment (laurit) [reply smoke-tests/src/test/groovy/io/opentelemetry/smoketest/JettyJpmsSmokeTest.groovy]: Thanks, you are absolutely right, this wasn't intentional.

**Pull request merged**